src/CoreWCF.Primitives/src/CoreWCF/Xml/Serialization/SoapSchemaExporter.cs

```diff
@@ -0,0 +1,478 @@
+        {
+            CheckScope(xmlMembersMapping.GetScope());
+            var membersMapping = xmlMembersMapping.GetAccessor().Mapping.AsMembersMappingSurrogate();
+            if (exportEnclosingType)
+            {
+                Fx.Assert("Not expecting to export enclosing type");
+                //ExportTypeMapping(membersMapping, null);
+            }
+            else
+            {
+                foreach (MemberMappingSurrogate memberMapping in membersMapping.Members)
+                {
+                    if (memberMapping.Elements.Length > 0)
+                        ExportTypeMapping(memberMapping.Elements[0].Mapping, null);
+                }
+            }
+        }
+
+        void CheckScope(object scope)
+        {
+            if (this.scope == null)
+            {
+                this.scope = scope;
+            }
+            else if (this.scope != scope)
+            {
+                throw new InvalidOperationException(SR.XmlMappingsScopeMismatch);
+            }
+        }
+
+        internal XmlDocument Document
+        {
+            get
+            {
+                if (document == null) document = new XmlDocument();
+                return document;
+            }
+        }
+
+        void CheckForDuplicateType(string newTypeName, string newNamespace)
+        {
+            XmlSchema schema = schemas[newNamespace];
+            if (schema != null)
+            {
+                foreach (XmlSchemaObject o in schema.Items)
+                {
+                    XmlSchemaType type = o as XmlSchemaType;
+                    if (type != null && type.Name == newTypeName)
+                        throw new InvalidOperationException(SR.Format(SR.XmlDuplicateTypeName, newTypeName, newNamespace));
+
+                }
+            }
+        }
+
+        void AddSchemaItem(XmlSchemaObject item, string ns, string referencingNs)
+        {
+            if (!SchemaContainsItem(item, ns))
+            {
+                XmlSchema schema = schemas[ns];
+                if (schema == null)
+                {
+                    schema = new XmlSchema();
+                    schema.TargetNamespace = ns == null || ns.Length == 0 ? null : ns;
+                    schema.ElementFormDefault = elementFormDefault;
+                    schemas.Add(schema);
+                }
+                schema.Items.Add(item);
+            }
+            if (referencingNs != null)
+                AddSchemaImport(ns, referencingNs);
+        }
+
+        void AddSchemaImport(string ns, string referencingNs)
+        {
+            if (referencingNs == null || ns == null) return;
+            if (ns == referencingNs) return;
+            XmlSchema schema = schemas[referencingNs];
+            if (schema == null) throw new InvalidOperationException(SR.Format(SR.XmlMissingSchema, referencingNs));
+            if (ns != null && ns.Length > 0 && FindImport(schema, ns) == null)
```

Comment:
`ns` is already null-checked on line 136

Reply:
Good catch